.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,33 @@
+name: run benchmarks
+on:
+  workflow_call:
+    inputs:
+      publish:
+        required: true
+        type: boolean
+      test-flags:
+        required: true
+        type: string
+jobs:
+  benchmarks:
+    runs-on: ubuntu-latest
+    steps:
+    - name: checkout
+      uses: actions/checkout@v2
+      with: 
+        fetch-depth: 1
+    - uses: actions/setup-go@v2
+      with: 
+        go-version: "1.20"
```

Comment:
What do you think about doing matrix go versions? (1.19 / 1.20)

Reply:
Because it will execute all benchmarks twice. I think we should use the latest available go version here to use all the performance improvements we can from Go itself. Why do you think it might be necessary to run using two Go versions?

Reply:
Ah, got it - no specific reason for needing to run it twice. Might be interesting to see if there is a bump between Go versions, but it's an unnecessary overhead now 🙃